### PR TITLE
feat: Add OR operation for Query Conditions in Alert module

### PR DESCRIPTION
**Pull request by uddhavdave**
This PR introduces a new enum `ConditionList` which allows setting complex conditions with grouping. It also takes care of backwards compatibility with its `LegacyConditions` variant which will be treated as a list of conditions which will be evaluated with `AND` operator. 

I have added tests for checking backwards compatibility and also test complex conditions such as 
```
{
  "query_condition": {
    "conditions": {
      "or": [
        {
          "and": [
            {
              "column": "level",
              "operator": "=",
              "value": "warning",
              "ignore_case": true
            },
            {
              "column": "job",
              "operator": "=",
              "value": "test",
              "ignore_case": false
            }
          ]
        },
        {
          "column": "level",
          "operator": ">",
          "value": "error",
          "ignore_case": false
        }
      ]
    }
  }
}
```

## Summary by CodeRabbit

- **Refactor**
  - Improved the handling and evaluation of alert conditions by introducing the `ConditionList` enum.
  - Updated the alert query-building logic to use the new `ConditionList` model.

- **Tests**
  - Added tests to ensure proper deserialization of various condition types in the alert system.

### CODE REVIEW

**Comment by uddhavdave** [comment]
Fixes #3047

**Comment by uddhavdave** [comment]
Ready for Review @oasisk @ansrivas

**Comment by ansrivas** [comment]
@uddhavdave could you please accept the CLA here? That will ensure that CI passes.

https://github.com/openobserve/openobserve/pull/3725#issuecomment-2162690351

File: src/service/alerts/mod.rs [hunk 1]
```diff
@@ -490,6 +487,89 @@
...
+                            "Column {} not found",
+                            &node.column,
+                        ));
+                    }
+                };
+                build_expr(node, "", data_type)
+            }
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (uddhavdave) [reply src/service/alerts/mod.rs:571]: Do we need to set some limit here? I think we should limit the number of `EndConditions` in a ConditionList in the business logic.

**Comment by hengfeiyang** [comment]
@uddhavdave Look good to me👍🏻

**Comment by uddhavdave** [comment]
Thanks @hengfeiyang 

Just pushed a small commit to improve comments. Should I mark it ready for review once CI passes?

**Pull request closed without merging**